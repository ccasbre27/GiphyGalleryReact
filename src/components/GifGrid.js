import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGrids } from '../hooks/useFetchGrids';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGrids( category );

    return (
        <div>
            <h3> { category } </h3>
            { loading && <p> Loading </p> }
            <div className="card-grid">
                <ol>
                    {
                        images.map( img => {
                            return <GifGridItem 
                                key={ img.id }
                                { ...img } />
                        })
                    }
                </ol>
            </div>
       </div>
    )
}
