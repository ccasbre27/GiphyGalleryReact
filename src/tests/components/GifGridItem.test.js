import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Tests on <GifGridItem />', () => {

    const title = 'My title';
    const url = 'https://localhost/algo.jpg';

    const wrapper = 
        shallow(<GifGridItem 
            title={ title }
            url={ url } /> );


    test('should render the component', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a paragraph with the title', () => {
       
        const paragraph = wrapper.find('p');
        
        expect( paragraph.text().trim() ).toBe( title );

    });

    test('should load the sent url to display proper image and fill the alt with the title', () => {
        
        const img = wrapper.find('img');
        
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });


    test('should have animate__fadeIn class', () => {
        
        const div = wrapper.find( 'div' );
        
        expect( div.prop('className') ).toContain('animate__fadeIn');
        
    })
    
    
    
        
});
