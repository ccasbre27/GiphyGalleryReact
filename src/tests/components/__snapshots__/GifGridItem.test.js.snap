// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> should render the component 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="My title"
    src="https://localhost/algo.jpg"
  />
  <p>
    My title
  </p>
</div>
`;
